abiencoder.go

```diff
@@ -399,23 +399,44 @@ func (a *ABI) writeField(binaryEncoder *Encoder, fieldName string, fieldType str
 }
 
 func valueToInt(fieldName string, value gjson.Result, bitSize int) (int64, error) {
-	i, err := strconv.ParseInt(value.Raw, 10, bitSize)
+	s := value.Raw
+	if len(s) > 0 && s[0] == '"' {
+		s = s[1:]
+	}
+	if len(s) > 0 && s[len(s)-1] == '"' {
+		s = s[:len(s)-1]
+	}
+	i, err := strconv.ParseInt(s, 10, bitSize)
```

Comment:
```suggestion
    // Explain here why we do this please
	i, err := strconv.ParseInt(strings.Trim(value.Raw, `"`), 10, bitSize)
```

This is exactly equivalent as the code you did with the same performance profile, trimming is a no-op in Go is the prefix/suffix is not present.